test/integration/render_pass.cc

```diff
@@ -399,6 +403,140 @@ void RenderPassTest::DepthGaussianNoise(const std::string &_renderEngine)
+
+  // create white material
+  MaterialPtr white = scene->CreateMaterial();
+  white->SetAmbient(0.5, 0.5, 0.5);
+  white->SetDiffuse(0.8, 0.8, 0.8);
+  white->SetReceiveShadows(true);
+  white->SetReflectivity(0);
+
+  // create plane
+  VisualPtr plane = scene->CreateVisual();
+  plane->AddGeometry(scene->CreatePlane());
+  plane->SetLocalScale(5, 8, 1);
+  plane->SetLocalPosition(3, 0, -0.5);
+  plane->SetMaterial(white);
+  root->AddChild(plane);
+
+  // capture original image with box (no distortion)
+  Image image = camera->CreateImage();
+  Image imageBarrel = camera->CreateImage();
+  Image imagePincushion = camera->CreateImage();
+  camera->Capture(image);
+
+  RenderPassSystemPtr rpSystem = engine->RenderPassSystem();
+  if (rpSystem)
+  {
+    // add barrel distortion pass
+    {
+      RenderPassPtr pass = rpSystem->Create<DistortionPass>();
+      DistortionPassPtr distortionPass =
+          std::dynamic_pointer_cast<DistortionPass>(pass);
+      distortionPass->SetK1(-0.1349);
+      distortionPass->SetK2(-0.51868);
+      distortionPass->SetK3(-0.001);
+      camera->AddRenderPass(distortionPass);
+      camera->Capture(imageBarrel);
+      camera->RemoveRenderPass(distortionPass);
+    }
+    // add pincushion distortion pass
+    {
+      RenderPassPtr pass = rpSystem->Create<DistortionPass>();
+      DistortionPassPtr distortionPass =
+          std::dynamic_pointer_cast<DistortionPass>(pass);
+      distortionPass->SetK1(0.1349);
+      distortionPass->SetK2(0.51868);
+      distortionPass->SetK3(0.001);
+      camera->AddRenderPass(distortionPass);
+      camera->Capture(imagePincushion);
+      camera->RemoveRenderPass(distortionPass);
+    }
+  }
+  else
+  {
+    ignwarn << "Engine '" << _renderEngine << "' does not support "
+            << "render pass  system" << std::endl;
+    return;
+  }
+
+  unsigned char* imageData = (unsigned char*)image.Data();
+  unsigned char* imageBarrelData = (unsigned char*)imageBarrel.Data();
+  unsigned char* imagePincushionData = (unsigned char*)imagePincushion.Data();
+
+  unsigned int colorSum = 0;
+  unsigned int colorSum2 = 0;
+  unsigned int colorSum3 = 0;
+  for (unsigned int y = 0; y < height; ++y)
+  {
+    for (unsigned int x = 0; x < width*3; x+=3)
+    {
+      unsigned int r = imageData[(y*width*3) + x];
+      unsigned int g = imageData[(y*width*3) + x + 1];
+      unsigned int b = imageData[(y*width*3) + x + 2];
+      colorSum += r + g + b;
+      unsigned int r3 = imageBarrelData[(y*width*3) + x];
+      unsigned int g3 = imageBarrelData[(y*width*3) + x + 1];
+      unsigned int b3 = imageBarrelData[(y*width*3) + x + 2];
+      colorSum2 += r3 + g3 + b3;
+      unsigned int r4 = imagePincushionData[(y*width*3) + x];
+      unsigned int g4 = imagePincushionData[(y*width*3) + x + 1];
+      unsigned int b4 = imagePincushionData[(y*width*3) + x + 2];
+      colorSum3 += r4 + g4 + b4;
+    }
+  }
+  EXPECT_GT(colorSum, colorSum2);
+  EXPECT_GT(colorSum, colorSum3);
```

Comment:
add a brief comment about this check? I think the distorted images are darker because of black borders?